script.js

```diff
@@ -31,6 +32,7 @@ function createGrid(gridNumber) {
                     cell.style.backgroundColor = color;
                 })
                 container.appendChild(cell)
+                cell.addEventListener('mouseover', changeColor);
```

Comment:
since these are being moved and added around, I added the event listener here. You might be able to get around this by doing event delegation (attaching an event to the parent and then checking which child is clicked):
https://www.freecodecamp.org/news/event-delegation-javascript/

That's handy because when html elements are added and removed via javascript, the events on them can disappear. However, if you add or remove the event to it's parent, the parent remains unchanging, and you can use that to check which of its children is clicked